microprofile/WFLY-13640_MicroProfile_Reactive_Messaging.adoc

```diff
@@ -0,0 +1,203 @@
+These changes are:
+<<<<<<< HEAD
+
+* The `Message` interface has new methods, and additionally a bunch of new factory methods (used both internally, and for users to wrap the payload in a `Message` instance from their `@Outgoing` annotated methods).
+* New classes:
+** `@Channel` annotation
+** `Emitter` interface - Only really makes sense if annotated with `@Channel`
+** `@OnOverflow` annotation
+** `@Metadata` class - Used by some of the new `Message` methods
+
+=======
+
+* The `Message` interface has new methods, and additionally a bunch of new factory methods (used both internally, and for users to wrap the payload in a `Message` instance from their `@Outgoing` annotated methods).
+* New classes:
+** `@Channel` annotation
+** `Emitter` interface - Only really makes sense if annotated with `@Channel`
+** `@OnOverflow` annotation
+** `@Metadata` class - Used by some of the new `Message` methods
+
+>>>>>>> b7a909b... Reduce scope for RM 1.0 and Kafka
+
+These changes are backward compatible, so the best we can do right now is to have users compile against the Reactive Messaging 1.0 API to 'hide' the new functionality from them. These will be included in the BOMs.
+
+Additionally, a deployment unit processor in the Reactive Messaging subsystem will by default ensure that there are no methods annotated with `@Channel` and `@Emitter` and throw an error if someone tries to use them from their deployment (for example if they compiled against the SmallRye Reactive Messaging API jar insteead of against the MicroProfile). The same applies to
+the usage of annotations from the `io.smallrye.reactive.messaging.annotations` package which are also considered experimental. Note that in WildFly, use of these will be allowed if the `jboss.as.reactive.messaging.experimental` system property is set to
+`true`, as this allows us to test the full functionality of each SmallRye Reactive Messaging upgrade. For product, this allowing
+of experimental features will be totally disabled.
+
+
+
+=== Organisation
+There is an extension/subsystem implementing each of the three specifications mentioned:
+
+* Reactive Messaging:
+** Extension Module: org.wildfly.extension.microprofile.reactive-messaging-smallrye
+** Subsystem Name: microprofile-reactive-messaging-smallrye
+* RSO:
+** Extension Module: org.wildfly.extension.microprofile.reactive-streams-operators-smallrye
+** Subsystem Name: microprofile-reactive-streams-operators-smallrye
+
+All subsystems are currently 'empty'. In other words they have no configuration. The presence of the subsystem turns on their respective functionality.
+
+=== Galleon Layers
+The functionality is split up into layers bringing in modules and subsystems.
+
+The layers involved are listed below:
+
+==== microprofile-reactive-messaging
+This brings in the MicroProfile Reactive Messaging subsystem, modules and associated functionality.
+
+Layer dependencies:
+
+* cdi
+* microprofile-config
+* microprofile-reactive-streams-operators
```

Comment:
I suppose that `cdi` dependency is mentioned here on purpose since the layer itself depends on it. Otherwise this is transitively brought in by the `microprofile-reactive-streams-operators` layer.

Reply:
Removed